python/cuml/ensemble/randomforest_common.pyx

```diff
@@ -285,9 +286,8 @@ class BaseRandomForestModel(Base):
             self.num_classes = len(self.classes_)
             for i in range(self.num_classes):
                 if i not in self.classes_:
-                    raise ValueError("The labels need "
-                                     "to be consecutive values from "
-                                     "0 to the number of unique label values")
+                    y_m, _ = make_monotonic(y_m)
+                    break
 
```

Comment:
I wonder if the logic for the loop might be better to have in `make_monotonic`, perhaps with a parameter like `make_monotonic(array, check_already_monotonic=True)` so that every use of the prim is cleaner? What do you think?

Reply:
I'd vote we make that change in RAFT, plumb `raft::label::make_monotonic` to Python (https://github.com/rapidsai/raft/issues/640), and then make a follow up PR to use that in cuML and remove this prim. 

Context: https://github.com/rapidsai/cuml/issues/4478#issuecomment-1118959085

What do you think?

Reply:
I did not modify the loop as I wanted to make least possible changes to the code. However, we can make use of check_lables primitives to see if the labels are already monotonic. I can make the change for cleaner implementation.

Reply:
@beckernick I am on board with that idea.